### PR TITLE
C#-friendly API

### PR DESCRIPTION
**Pull request by anton-pt**
Here is an initial example of a C#-friendly API. I fully expect some tweaks to be required before it goes live, and probably some more testing. Currently, there is a demo C# WPF project in the tests folder, which is admittedly not the best place to put that but I just wanted to get the discussion rolling.

What are your thoughts on the C# wrapper? Do you think the C# and F# APIs are reasonably isolated? I aimed to keep everything the same for F# provided that you use `open FSharp.ViewModule` and then in C# if you use `using CSharp.ViewModule` then all the F# APIs are hidden but the new C# APIs are there. There's not much testing currently on the F# side, do you think it's necessary to add something on the C# side?

### CODE REVIEW

**Comment by ReedCopsey** [comment]
should we do this as a separate branch for now?

**Comment by anton-pt** [comment]
Sounds like a good plan. Do you need me to submit the PR differently in order to do that? I'm not entirely sure why the build failed on the CI server,... worked fine locally but it seems like maybe it can't find a DLL or something.

Also perhaps it would be good to publish to NuGet with a 'csharp' pre-release tag just so it's easy to test it out in some projects and make sure I haven't broken anything on the F# side.

File: src/FSharp.ViewModule/Validation.fs [hunk 1]
```diff
@@ -150,66 +150,73 @@ module Validators =
         workflow(value)
 
 
-namespace CSharp.ViewModule.Validation
+namespace ViewModule.CSharp.Validation
```
Comment (ReedCopsey) [inline src/FSharp.ViewModule/Validation.fs]: Why the change from order (see line 17)?

File: paket.lock [hunk 2]
```diff
@@ -1,26 +1,25 @@
...
+    Microsoft.Bcl.Build (1.0.21) - import_targets: false, framework: net10, net11, net20, net30, net35, net40, net40-full
+    Microsoft.Net.Http (2.2.29) - framework: net10, net11, net20, net30, net35, net40, net40-full
       Microsoft.Bcl (>= 1.1.10)
       Microsoft.Bcl.Build (>= 1.0.14)
     NUnit (2.6.4)
     NUnit.Runners (2.6.4)
-    Octokit (0.17.0)
-      Microsoft.Net.Http
-    SourceLink.Fake (1.1.0)
+    Octokit (0.21.1)
```
Comment (ReedCopsey) [inline paket.lock]: Can you revert the Octokit changes in the lock (or at least verify that it doesn't break doc generation?) I've had issues with the newer versions breaking doc generation

Comment (JohanLarsson) [inline paket.lock]: Is there a way to pin the octokit dependency?

Comment (forki) [inline paket.lock]: what has octokit to do with docs`?

Comment (ReedCopsey) [inline paket.lock]: @forki no idea, but not fixing it seems to break the doc generation in project scaffold.  No idea why.

**Comment by anton-pt** [comment]
I just noticed that there seems to be an issue when overriding Validate in C# to create a composite validator. Even after correcting the error, IsValid seems to remain false. Strangely still seems to work in F# with the FsXaml demos though.

So current TODO is:
- fix override Validate bug (it may only be in the C# demo application)
- add C# API for progress reporting 
- add C# extension methods to run validators directly so that you can use them inside override Validate like the FsXaml example.
